wasp-ai/src/client/pages/ResultPage.jsx

```diff
@@ -406,59 +413,71 @@ function getCardinalNumber(number) {
   }
 }
 
-export function StarOurRepoModal({ isStarRepoOpen, setIsStarRepoOpen, appGenerationResult }) {
-  const [tokens, setTokens] = useState(0)
-  const { data: numTotalProjects } = useQuery(getNumProjects, {}, { enabled: isStarRepoOpen })
-
-  const tokenNumberStr = appGenerationResult?.project?.logs?.filter(log => log.content.toLowerCase().includes("tokens usage") === true)[0]?.content.split(':')[1]?.trim()
+export function OnSuccessModal({ isOpen, setIsOpen, appGenerationResult }) {
+  const [numTokensSpent, setNumTokensSpent] = useState(0)
+  const { data: numTotalProjects } = useQuery(getNumProjects, {}, { enabled: isOpen })
 
   useEffect(() => {
-    if (!!tokenNumberStr) {
-      const num = tokenNumberStr.slice(1, -1) * 1000
-      setTokens(num)
+    const logText = appGenerationResult?.project?.logs?.find((log) => /tokens usage/i.test(log.content))?.content;
+    const tokenNumberStr = logText ? logText.split(":")[1]?.trim() : null;
+    if (tokenNumberStr) {
+      const num =  parseInt(tokenNumberStr.replace(/[\s~k]/gi, '')) * 1000;
+      if (num.toString().match(/^[0-9]+$/)) {
```

Comment:
Hm unusual that this is how you would need to check if `num` is actual int, I would expect there to be a better way. But if you do change from comment above, this should also simplify / dissapear.